Assets/MixedRealityToolkit/MSBuild/Scripts/CompilationPlatformInfo.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#if UNITY_EDITOR
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEditor.Compilation;
+
+namespace Microsoft.MixedReality.Toolkit.MSBuild
+{
+    /// <summary>
+    /// This contains parsed information using Unity configuration about a specific compilation platform.
+    /// </summary>
+    public class CompilationPlatformInfo
+    {
+        /// <summary>
+        /// Given a non-editor <see cref="AssemblyDefinitionPlatform"/> platform, creates an instances of CompilationPlatform fetching defines and references.
+        /// </summary>
+        /// <param name="platform">The platform to use for parsing.</param>
+        /// <returns>The <see cref="CompilationPlatformInfo"/> containing building infromation for the platform.</returns>
+        public static CompilationPlatformInfo GetCompilationPlatform(AssemblyDefinitionPlatform platform)
+        {
+            if (platform.BuildTarget == BuildTarget.NoTarget)
+            {
+                throw new ArgumentException(nameof(platform), "Provided an editor platform, use GetEditorPlatform.");
+            }
+
+            AssemblyBuilder builder = new AssemblyBuilder("dummy.dll", new string[] { @"Editor\dummy.cs" })
+            {
+                buildTarget = platform.BuildTarget,
+                buildTargetGroup = Utilities.GetBuildTargetGroup(platform.BuildTarget),
+                flags = AssemblyBuilderFlags.None
+            };
+
+            HashSet<string> platformCommonDefines = new HashSet<string>(builder.defaultDefines);
+            HashSet<string> playerDefines = new HashSet<string>(builder.defaultDefines);
+
+            HashSet<string> platformCommonReferences = new HashSet<string>(FilterOutProjectReferences(builder.defaultReferences));
+            HashSet<string> playerReferences = new HashSet<string>(FilterOutProjectReferences(builder.defaultReferences));
+
+            builder.flags = AssemblyBuilderFlags.EditorAssembly;
```

Comment:
>builder.flags = AssemblyBuilderFlags.EditorAssembly; [](start = 12, length = 52)

Can you add a comment for why this has to be set after the creation of the builder? A naive reading of this would maybe have a person just move this to the constructor.

Reply:
But clearly that would be bad, otherwise it wouldn't have been done this way.

---
In reply to: [301785013](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5215#discussion_r301785013) [](ancestors = 301785013)

Reply:
Ah, I guess what's weird to me is that setting the property actually changes the hashsets of other properties, I guess that was what was throwing me off. Nevermind!

---
In reply to: [301785122](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5215#discussion_r301785122) [](ancestors = 301785122,301785013)

Reply:
Will add comment.

---
In reply to: [301785751](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5215#discussion_r301785751) [](ancestors = 301785751,301785122,301785013)